sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -48,6 +48,29 @@
  */
 public class V2ExpressionSQLBuilder {
 
+  protected String escapeSpecialChar(String str) {
+    StringBuilder builder = new StringBuilder();
+
+    for (char c : str.toCharArray()) {
+      switch (c) {
+        case '_':
+          builder.append("\\_");
+          break;
+        case '%':
+          builder.append("\\%");
+          break;
+        case '\'':
+          builder.append("\'");
```

Comment:
```suggestion
          builder.append("\\\'");
```

Reply:
H2 doesn't it.

```
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "SELECT ""EMAIL"" FROM ""test"".""address""  WHERE (""EMAIL"" IS NOT NULL) AND (""EMAIL"" LIKE 'abc\\_\\'[*]\\'\\%%')    "; SQL statement:
SELECT "EMAIL" FROM "test"."address"  WHERE ("EMAIL" IS NOT NULL) AND ("EMAIL" LIKE 'abc\_\'\'\%%')     [42000-220]
```

Reply:
For `$"email".startsWith("abc_'%")`, spark SQL will escape the single quote.
So the DS V2 will get the input `abc_''%`.
H2 also supports use `'` to escape `'`.

Reply:
> spark SQL will escape the single quote.

Where does it happen?

Reply:
`JDBCSQLBuilder.compileValue` calls `escapeSql`.
https://github.com/apache/spark/blob/0b7a537be557032abddf42f143a079889a584c18/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala#L289

Reply:
ah good to know. Then we can remove `case '\'':`